doc/source/whatsnew/v2.0.1.rst

```diff
@@ -32,7 +32,7 @@ Bug fixes
 
 Other
 ~~~~~
--
+- Bug in :class:`Index` where creating or converting to the numpy dtype S3 would raise ``NotImplementedError`` (:issue:`50127`)
```

Comment:
instead of specifically S3 should this be numpy string dtypes more generally?

Reply:
Done.